### PR TITLE
Use symlinks instead of bind mounts for mysqlrouter directories

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
## Issue
Directories that use bind mounts cannot be deleted & re-created (they will not be remounted)
The mysql-router vm charm deletes directories to clean up configuration & state when the mysql<->mysqlrouter relation is broken

## Solution
Use symlinks instead of bind mounts

## Context
https://forum.snapcraft.io/t/snap-layouts/7207/33

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -29,7 +25,7 @@ jobs:
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
         with:
           snap: ${{ steps.get-snap-name.outputs.SNAP_FILE }}
-          release: ${{ github.ref_name }}
+          release: 8.0/edge/pr-33
```
Comment (shayancanonical) [inline .github/workflows/publish.yaml]: is this a leftover?

**Review by shayancanonical — APPROVED** [review]

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (carlcsaposs-canonical) [reply .github/workflows/publish.yaml]: yes; will remove before merging

**Review by paulomach — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]

**Comment by taurus-forever** [comment]
Q: will it affect OCI somehow?

**Comment by carlcsaposs-canonical** [comment]
> Q: will it affect OCI somehow?

It shouldn't—the OCI directories like `/etc/mysqlrouter` are at the actual path `/etc/mysqlrouter`, not nested inside a folder as it is in the snap

**Pull request merged**